burn-tensor/src/tensor/shape.rs

```diff
@@ -14,6 +14,24 @@ impl<const D: usize> Shape<D> {
 
         num_elements
     }
+
```

Comment:
We need documentation for this public API. If you could also mention why we need this (a typical use), it'd be great. Also having some demonstrative formula for calculation is helpful.